sdk/cosmos/azure-cosmos/azure/cosmos/_base.py

```diff
@@ -673,3 +676,14 @@ def validate_cache_staleness_value(max_integrated_cache_staleness):
     int(max_integrated_cache_staleness)  # Will throw error if data type cant be converted to int
     if max_integrated_cache_staleness <= 0:
         raise ValueError("Parameter 'max_integrated_cache_staleness_in_ms' can only be a positive integer.")
+
+
+def auto_scale_header(auto_scale):
```

Comment:
This feels weird as a method. Also, I'd recommend adding type hints since I don't know what `auto_scale` is. Now that we no longer support Python 2, you can use type hints in both sync and async and for new stuff I would argue you definitely should.

Reply:
Yeah, I don't think you need this method at all. 
`str(auto_scale.__dict__)` should do what you need, I would think.

Reply:
Hello Travis the reason why we needed to use this method is because we are using a nested dictionary and when doing it the way you are thinking will cause it not to work as it just places both properties in a single dictionary and we are still missing dictionaries that aren't directly linked to our object, I have linked an example image showing what it is supposed to look like. 
![Screenshot 2022-06-23 113218](https://user-images.githubusercontent.com/51177847/175416649-1d85150b-077c-46f3-9796-525919ef0096.png)

Reply:
I don't understand. What is is you expect the payload to look like?

Reply:
![Screenshot 2022-06-30 111929](https://user-images.githubusercontent.com/51177847/176751274-cbfd16ba-4487-46c6-8f5c-790e3842937f.png)
This is what it is expected to look like but when doing it the way you are thinking, it will not work because we are not using the layered dictionary, your version will look like this ![Screenshot 2022-06-30 112414](https://user-images.githubusercontent.com/51177847/176751598-291a5c5f-932f-4696-9bb2-5eaf8dc44a09.png) and it will not run because we are not getting any of our values from the properties

Reply:
Why is the input format different from the format needed on the wire? I just don't know of any other place in the Python SDK where we do something like this, so it seems immediately suspect. But @annatisch has a lot more experience in the Python SDK than me, so I'll gladly defer to her on this.